biblib/views/operations_view.py

```diff
@@ -0,0 +1,359 @@
+                intersect_lib = intersect_lib.intersection(set(secondary_library.get_bibcodes()))
+
+        if len(intersect_lib) < 1:
+            current_app.logger.info('No records remain after taking the intersection of {0} and {1}'
+                                    .format(library_id, document_data['libraries']))
+        return list(intersect_lib)
+
+    @classmethod
+    def diff_libraries(cls, library_id, document_data):
+        """
+        Takes the difference of two or more libraries
+        :param library_id: the primary library ID
+        :param document_data: dict containing the list 'libraries' that holds the secondary library IDs
+
+        :return: list of bibcodes in the differenced set
+        """
+        current_app.logger.info('User requested to take the difference between {0} with {1}'
+                                .format(library_id, document_data['libraries']))
+        with current_app.session_scope() as session:
+            # Find the specified library
+            primary_library = session.query(Library).filter_by(id=library_id).one()
+            diff_lib = set(primary_library.get_bibcodes())
+
+            for lib in document_data['libraries']:
+                if isinstance(lib, basestring):
+                    lib = cls.helper_slug_to_uuid(lib)
+                secondary_library = session.query(Library).filter_by(id=lib).one()
+                diff_lib = diff_lib.difference(set(secondary_library.get_bibcodes()))
+
+        if len(diff_lib) < 1:
+            current_app.logger.info('No records remain after taking the difference between {0} and {1}'
+                                    .format(library_id, document_data['libraries']))
+        return list(diff_lib)
+
+    @classmethod
+    def copy_library(cls, library_id, document_data):
+        """
+        Copies the contents of one library into another
+        :param library_id: primary library ID, library to copy
+        :param document_data: dict containing the list 'libraries' which holds one secondary library ID; this is
+            the library to copy over
+
+        :return: dict containing the metadata of the copied-over library (the secondary library)
+        """
+        current_app.logger.info('User requested to copy the contents of {0} into {1}'
+                                .format(library_id, document_data['libraries']))
+
+        secondary_libid = document_data['libraries'][0]
+        if isinstance(secondary_libid, basestring):
+            secondary_libid = cls.helper_slug_to_uuid(secondary_libid)
+
+        metadata = {}
+        with current_app.session_scope() as session:
+            primary_library = session.query(Library).filter_by(id=library_id).one()
+            good_bib = primary_library.get_bibcodes()
+
+            secondary_library = session.query(Library).filter_by(id=secondary_libid).one()
+            secondary_library.remove_bibcodes(secondary_library.get_bibcodes())
```

Comment:
I'm of two minds about this. Yes, the functionality you're describing would reduce errors. However, if we don't empty the second library first, then we're basically just taking the union of the two libraries and don't need this copy function after all.

Reply:
But the `union` creates a new library, the `copy` copies the content of an existing library into another existing library and it would be up to the user to empty that target library before copying. Imagine the `cp` command in unix, if you copy the content of a directory to another directory, you would not expect the target directory to be cleaned first. Does this make sense?